RadixWallet/Features/TransactionReviewFeature/TransactionReviewAccount/TransactionReviewAccount+View.swift

```diff
@@ -96,134 +93,64 @@ extension TransactionReviewAccount {
 
 // MARK: - TransactionReviewResourceView
 struct TransactionReviewResourceView: View {
-	let transfer: TransactionReview.Transfer
+	let transfer: ResourceBalance // FIXME: GK use viewstate
 	let onTap: (OnLedgerEntity.NonFungibleToken?) -> Void
 
 	var body: some View {
 		switch transfer.details {
-		case let .fungible(details):
-			TransactionReviewFungibleView(viewState: .init(resource: transfer.resource, details: details), background: .app.gray5) {
-				onTap(nil)
-			}
-		case let .nonFungible(details):
-			TransferNFTView(viewState: .init(resource: transfer.resource, details: details), background: .app.gray5) {
-				onTap(nil)
-			}
-		case let .liquidStakeUnit(details):
-			LiquidStakeUnitView(viewState: .init(resource: transfer.resource, details: details), background: .app.gray5) {
-				onTap(nil)
-			}
-		case let .poolUnit(details):
-			PoolUnitView(viewState: .init(resource: transfer.resource, details: details), background: .app.gray5) {
+		case .fungible, .nonFungible, .liquidStakeUnit, .poolUnit:
+			ResourceBalanceButton(transfer.viewState, appearance: .transactionReview) {
 				onTap(nil)
 			}
 		case let .stakeClaimNFT(details):
-			StakeClaimResourceView(viewState: details, background: .app.gray5) { stakeClaim in
+			ResourceBalanceView.StakeClaimNFT(viewState: details, background: .app.gray5, compact: false) { stakeClaim in
 				onTap(stakeClaim.token)
 			}
 		}
 	}
 }
 
-// MARK: - TransactionReviewAmountView
-struct TransactionReviewAmountView: View {
-	let amount: RETDecimal
-	let guaranteedAmount: RETDecimal?
-
-	var body: some View {
-		VStack(alignment: .trailing, spacing: 0) {
-			if guaranteedAmount != nil {
-				Text(L10n.TransactionReview.estimated)
-					.textStyle(.body2HighImportance)
-					.foregroundColor(.app.gray1)
-			}
-			Text(amount.formatted())
-				.textStyle(.body1Header)
-				.foregroundColor(.app.gray1)
-
-			if let guaranteedAmount {
-				Text(L10n.TransactionReview.guaranteed)
-					.textStyle(.body2HighImportance)
-					.foregroundColor(.app.gray2)
-					.padding(.top, .small3)
-
-				Text(guaranteedAmount.formatted())
-					.textStyle(.body1Header)
-					.foregroundColor(.app.gray2)
-			}
+extension [ResourceBalance.ViewState.Fungible] { // FIXME: GK use full
```

Comment:
FIXME comment still relevant?

Reply:
Same here, or rather here it's exactly the opposite thing.